File: 101/exercise_rework/easy_5/bannerizer.rb
# bannerizer.rb

def outer(lng)
  "+-#{'-' * lng}-+"
end

def inner(lng, contents)
  return "| #{contents} |" unless contents.empty?
  "| #{' ' * lng} |"
end

def build_box(str)
  content_length = str.length
  box = ''
  box << outer(content_length) + "\n"
  box << inner(content_length, '') + "\n"
  box << inner(content_length, str) + "\n"
  box << inner(content_length, '') + "\n"
  box << outer(content_length)
end

def print_in_box(str)
  puts build_box(str)
end

print_in_box('To boldly go where no one has gone before.')

File: 210/lesson_3/logic_and_flow/small_problems/easy2/bannerizer.js
// bannerizer.js

function multiplyCharacter(char, times) {
  var repeatedChars = '';
  for (var i = 0; i < times; i += 1) {
    repeatedChars += char;
  }
  return repeatedChars;
}

function logOuter(len) {
  var lineStr = multiplyCharacter('-', len);
  console.log(`+-${lineStr}-+`)
}

function logInner(len) {
  var cushionStr = multiplyCharacter(' ', len);
  console.log(`| ${cushionStr} |`)
}

function logInBox(str) {
  var strLength = str.length;
  logOuter(strLength);
  logInner(strLength);
  console.log(`| ${str} |`);
  logInner(strLength);
  logOuter(strLength);
}

logInBox('To boldly go where no one has gone before.');
logInBox('');